From 4e3e2604c816e068b8050b69d3dd082ee035460d Mon Sep 17 00:00:00 2001
From: Rogerio Nunes <rogerio.nunes@nxp.com>
Date: Thu, 28 Jul 2016 20:25:55 -0400
Subject: add pkg-config file support

Upstream status: pending

Signed-off-by: Rogerio Nunes <rogerio.nunes@nxp.com>

diff --git a/.gitignore b/.gitignore
index 8115d26..1b72b00 100644
--- a/.gitignore
+++ b/.gitignore
@@ -59,3 +59,4 @@ Makefile
 /m4
 /missing
 /stamp-h1
+/libnxp_nfc.pc
diff --git a/Makefile.am b/Makefile.am
index 512fad3..243aec0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,6 +25,7 @@ AM_CPPFLAGS += \
 endif
 	
 configdir = ${sysconfdir}
+pkgconfiglibdir = $(libdir)/pkgconfig
 
 if PN7120
 config_DATA = conf/PN7120/libnfc-brcm.conf conf/PN7120/libnfc-nxp.conf
@@ -34,6 +35,10 @@ if PN7150
 config_DATA = conf/PN7150/libnfc-brcm.conf conf/PN7150/libnfc-nxp.conf
 endif
 
+pkgconfiglib_DATA = \
+	libnxp_nfc.pc
+
+
 LIBNFC_NCI_INCLUDE := \
 	src/libnfc-nci/include \
 	src/libnfc-nci/gki/ulinux \
diff --git a/configure.ac b/configure.ac
index f4e1bfd..242bf76 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.69])
-AC_INIT([nxp_nfc], [2.2])
+AC_INIT([nxp_nfc], [2.2], [], [], [https://github.com/NXPNFCLinux/linux_libnfc-nci])
 AM_INIT_AUTOMAKE([foreign subdir-objects color-tests])
 AC_CONFIG_HEADERS([config.h])
 
@@ -54,7 +54,10 @@ AC_CHECK_FUNCS([select])
 AC_CHECK_FUNCS([dlopen])
 AC_CHECK_FUNCS([strtol])
 
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([
+Makefile
+libnxp_nfc.pc
+])
 
 AC_ARG_ENABLE([pn7150],
 [  --enable-pn7150    set chip version to PN7150],
@@ -92,13 +95,17 @@ if [test "$COUNT" -gt 1]; then
 AC_MSG_ERROR(Can not enable multiple chip type)
 fi])
 
+LIBNXP_NFC_PRIVATE_PKGS=""
+
 AM_CONDITIONAL([LLCP1_3],  [test x$llcp1_3  = xtrue])
-AM_COND_IF([LLCP1_3], [
-  PKG_CHECK_MODULES([OPENSSL], [openssl], [
-    AC_MSG_NOTICE([LLCP1.3 support is enabled (secured P2P)])], [
-    AC_MSG_ERROR(openssl not available but required when LLCP1.3 support is enabled)])], [
-  AC_MSG_NOTICE([LLCP1.3 support is disabled (no secure P2P)])
-])
+AM_COND_IF([LLCP1_3],
+  [PKG_CHECK_MODULES([OPENSSL], [openssl],
+    [AC_MSG_NOTICE([LLCP1.3 support is enabled (secured P2P)])
+    LIBNXP_NFC_PRIVATE_PKGS="openssl"],
+    [AC_MSG_ERROR(openssl not available but required when LLCP1.3 support is enabled)])],
+  [AC_MSG_NOTICE([LLCP1.3 support is disabled (no secure P2P)])])
+
+AC_SUBST(LIBNXP_NFC_PRIVATE_PKGS)
 
 AM_CONDITIONAL([CHIP_SEL_ONE], [if [$pn7150]; then
 AC_MSG_NOTICE([Selected chip type is PN7150])
diff --git a/libnxp_nfc.pc.in b/libnxp_nfc.pc.in
new file mode 100644
index 0000000..3dd8ea0
--- /dev/null
+++ b/libnxp_nfc.pc.in
@@ -0,0 +1,12 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: libnxp_nfc
+Description: NXP NFC library
+Version: @VERSION@
+URL: @PACKAGE_URL@
+Requires.private: @LIBNXP_NFC_PRIVATE_PKGS@
+Cflags: -I${includedir}
+Libs: -L${libdir} -lnxp_nfc
-- 
1.9.1

